### ISSUE TITLE
Some SoundSourceProxyTests fail on Win11 with VS2022

### ISSUE THREAD

**Issue by JoergAtGithub** [mixxxdj/mixxx#11094]
### Bug Description

I build the same versions of Main ([9965d55](https://github.com/mixxxdj/mixxx/commit/9965d55f6e00cd1982f78cf09ac6196887869487)) and PR #11074 on two Windows laptops. On the older one with Win7 and VS2019 all tests passed.
But on the newer one with Win11 and VS2022 four tests failed:
```
SoundSourceProxyTest.seekForwardBackward 
SoundSourceProxyTest.seekBoundaries
SoundSourceProxyTest.readBeyondEnd
SoundSourceProxyTest.regressionTestCachingReaderChunkJumpForward
```

The test SoundSourceProxyTest.seekForwardBackward generated ~2GByte of similar log messages, here is shortened version of the log file:
[LastTest_soundsourceproxytestsfailed_newbuilenv.log](https://github.com/mixxxdj/mixxx/files/10098214/LastTest_soundsourceproxytestsfailed_newbuilenv.log)

### Version

Main

### OS

Windows11 + VS2022

**Comment by uklotzde** [comment]
`SoundSourceMediaFoundation` is hardly working due to lack of Windows developers/testers and might fail when Microsoft decides to either change the API or behavior.

The API itself is not really suitable for sample-accurate decoding.

**Comment by uklotzde** [comment]
https://www.reddit.com/r/DJs/comments/z9f7yf/comment/iygulke/?utm_source=share&utm_medium=web2x&context=3

**Comment by JoergAtGithub** [comment]
I just updated to the latest Windows11 update - these tests still fail.

**Comment by daschuer** [comment]
Without rebuilding?

**Comment by JoergAtGithub** [comment]
I now copied all .exe and .dll files from the build directory of my Windows7 laptop to the build directory of my Windows11 laptopand ran CTest.
On Windows7 all tests with these executables were pass, on Windows11 the 4 SoundSourceProxyTests failed.

Conclusion: It's an unfixed OS bug in Windows11 and not a Mixxx bug.

**Comment by JoergAtGithub** [comment]
These tests still fail, using latest Win11.

**Comment by daschuer** [comment]
@JoergAtGithub: Does Windows 11 find the Sound Start Cue created in Windows < 11? Just play one or more m4a file you had in your Windows before the OS Update. 

and look for `First sound has been moved!`

Than please Build Mixxx with ffmpeg 
https://github.com/mixxxdj/mixxx/blob/a547664420463515926a619aafe08b378b34e035/src/sources/soundsourceffmpeg.cpp#L470 = Highest 

And verify it again.

**Comment by daschuer** [comment]
The Test failure is reproducible on the GitHub Workflows using Windows-2022:
https://github.com/daschuer/mixxx/actions/runs/5855464923/job/15873332041

**Comment by JoergAtGithub** [comment]
> The Test failure is reproducible on the GitHub Workflows using Windows-2022

That means that also newer releases of the Windows10 series are affected: https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md

**Comment by daschuer** [comment]
As expected the windows-2022 build with FFmpeg succeeds https://github.com/daschuer/mixxx/actions/runs/5865154501/job/15901524981

The crucial question is if the MediaFoundation Windows 10 decoded files match FFmpeg decoded files perectly, or if we need to consider a cue point shift. 

@JoergAtGithub: did you found time to check for the "First sound has been moved!" log entry with m4a files? Interesting would bee if you see that in the FFmpeg build linked above. 

I think we should establish a unit test for it.

**Comment by daschuer** [comment]
To continue here we need a Windows user with a Windows 11 machine that has a significant amount (> 10) m4a tracks analysed with WIN10 or before.  Who can help here?

**Comment by JoergAtGithub** [comment]
I could to the analysis task, since I've both, a Win7 and Win11 laptop with latest Mixxx builds. But I don't have any m4a tracks. Does anybody know, where I could download suitable test files?

**Comment by daschuer** [comment]
Thank you. You can just convert a view. I think Audacity can do it. It uses Ffmpeg.

**Comment by daschuer** [comment]
It still fails with Windows 2022: 
https://github.com/daschuer/mixxx/actions/runs/7285956895/job/19853789720

**Comment by daschuer** [comment]
The Preview Windows 10 is also failing :-(

10.0.20348.2031fail  
10.0.20348.1fail 
10.0.19041.3636 fail

10.0.17763.2989 OK

**Comment by daschuer** [comment]
Tested with: https://github.com/mixxxdj/mixxx/issues/12289#issuecomment-1867834433

**Comment by daschuer** [comment]
Do we have a working Windows version in the meantime?

**Comment by JoergAtGithub** [comment]
No, this test is not fixed in any Windows 11 version known to me.

**Comment by EchoCoder1729** [comment]
How are we planning to fox this? Any leads so that I can take this up and maybe fix this test? I am ready to dive into the Windows 11 Sound Driver API calls if necessary. But if this test is going to be breaking across windows versions, then doesn't make much sense to work on it though. Just to clarify, each platform (Linux/ Windows) uses their own test suites or do they run the same tests?

**Comment by JoergAtGithub** [comment]
The soundsource code tested here is platform specific. In general we have only one test suite.

**Comment by daschuer** [comment]
I have just pushed a new CI build to check the current situation with Windows 2022 
https://github.com/daschuer/mixxx/actions/runs/10733009450

It is Windows Server 2022, Version 21H2: 10.0.20348 Build 2655
https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md

**Comment by EchoCoder1729** [comment]
Do we have any environment variable to specify that it is a windows machine and disable the specific test in case windows? Is it that we need to modify the test or any changes would break the test in previous versions like windows 7?

**Comment by daschuer** [comment]
Windows is really broken here. So the failure of the test is correct. You may disable/skip the tesz if you like.

**Comment by daschuer** [comment]
Since https://github.com/mixxxdj/mixxx/pull/14658 is merged, we can more easy visualise the issue using the csv output from the failing test output: 

```
2026-09-15T15:37:20.5412276Z CSV debug output with 9740 errors in the range 0 to 9999:
2026-09-15T15:37:20.5412374Z sep=,
2026-09-15T15:37:20.5412697Z expected left,expected right,actual left, actual right, error
2026-09-15T15:37:20.5413332Z 0.158874512,0.158874512,-0.673309326,-0.673309326,1
2026-09-15T15:37:20.5413639Z 0.0854492188,0.0854492188,-0.706878662,-0.706878662,1
2026-09-15T15:37:20.5413958Z 0.00961303711,0.00961303711,-0.732299805,-0.732299805,1
```

SoundSourceProxyTest.seekForwardBackward
with "id3-test-data/cover-test-��~l��-ffmpeg-aac.m4a" 
using provider "Microsoft Media Foundation 10.0.26100.33296"

Fails with an offset of 78 frames = ~1.8 ms 
A beat offset is hear-able from 25 ms  

<img width="1592" height="392" alt="Image" src="https://github.com/user-attachments/assets/2aa1e6f3-46c0-4389-a1cd-f4429556588c" />

The next step to fix the issue is to apply these offset after seeking conditionally depending in the Microsoft Media Foundation version.  
I am however not sure if this is a constant. 

I am not on Windows so I will not continue here. 

Any volunteers? 

Since it will unlikely happen before the 2.6 release I will remove it form the milestone.
